src/dbnode/integration/setup.go

```diff
@@ -574,6 +554,91 @@ func (ts *testSetup) sleepFor10xTickMinimumInterval() {
 	time.Sleep(opts.TickMinimumInterval() * 10)
 }
 
+func (ts *testSetup) httpClusterAddr() string {
+	if addr := ts.opts.HTTPClusterAddr(); addr != "" {
+		return addr
+	}
+	return *httpClusterAddr
+}
+
+func (ts *testSetup) httpNodeAddr() string {
+	if addr := ts.opts.HTTPNodeAddr(); addr != "" {
+		return addr
+	}
+	return *httpNodeAddr
+}
+
+func (ts *testSetup) tchannelClusterAddr() string {
+	if addr := ts.opts.TChannelClusterAddr(); addr != "" {
+		return addr
+	}
+	return *tchannelClusterAddr
+}
+
+func (ts *testSetup) tchannelNodeAddr() string {
+	if addr := ts.opts.TChannelNodeAddr(); addr != "" {
+		return addr
+	}
+	return *tchannelNodeAddr
+}
+
+func (ts *testSetup) httpDebugAddr() string {
+	if addr := ts.opts.HTTPDebugAddr(); addr != "" {
+		return addr
+	}
+	return *httpDebugAddr
+}
+
+func (ts *testSetup) maybeResetClients() error {
```

Comment:
Super nit: Maybe just call this `resetClients()`?  Kind of implementation detail that it won't if it wasn't set.